cg/meta/transfer/external_data.py

```diff
@@ -45,32 +45,51 @@ def create_log_dir(self, dry_run: bool, ticket_id: int) -> Path:
         log_dir.mkdir(parents=True, exist_ok=False)
         return log_dir
 
-    def get_source_path(self, cust_sample_id: str, customer: str, ticket_id: int) -> Path:
+    def get_source_path(
+        self,
+        customer: str,
+        ticket_id: int,
+        cust_sample_id: Optional[str] = "",
+    ) -> Path:
         """Returns the path to where the sample files are fetched from"""
         return Path(self.caesar_path % customer, str(ticket_id), cust_sample_id)
 
     def get_destination_path(self, customer: str, lims_sample_id: str) -> Path:
         """Returns the path to where the files are to be transferred"""
         return Path(self.hasta_path % customer, lims_sample_id)
 
-    def transfer_sample(
-        self, cust: str, cust_sample_id: str, lims_sample_id: str, dry_run: bool, ticket_id: int
-    ) -> int:
-        """Runs a SLURM job to transfer a sample folder. Returns SLURM jobid of the transfer process"""
+    def transfer_sample_files_from_source(self, dry_run: bool, ticket_id: int):
+        """Transfers all sample files, related to given ticket, from source to destination"""
+        samples: List[models.Sample] = self.status_db.get_samples_from_ticket(ticket_id=ticket_id)
+        cust: str = self.status_db.get_customer_id_from_ticket(ticket_id=ticket_id)
+        available_samples: List[models.Sample] = [
+            sample
+            for sample in samples
+            if sample.name
+            in [
+                sample_folder.parts[-1]
+                for sample_folder in self.get_source_path(
+                    customer=cust, ticket_id=ticket_id
+                ).iterdir()
+            ]
+        ]
```

Comment:
```suggestion
        available_samples: List[models.Sample] = [
            sample
            for sample in samples
            if sample.name
            in [
                sample_folder.parts[-1]
                for sample_folder in self.get_source_path(
                    customer=cust, ticket_id=ticket_id
                ).iterdir()
            ]
        ]
```
Could you make a "get_available_samples_from_source_path" def so that the code is more readable, reusable and you can write unit tests to make sure that the "for" and "ifs" do what they should?

Reply:
Agreed. This was done in the next PR but I'll move it here